### ISSUE TITLE
execCommand("indent") conflicts with the content model of <ul>/<ol>

### ISSUE THREAD

**Issue by saschanaz** [w3c/editing#256]
Consider the following tree:

```html
<ul>
<li>first</li>
<li>second</li>
</ul>
```

Here, `execCommand("indent")` on the second list item generates the following tree:

```html
<ul>
<li>first</li>
<ul><li>second</li></ul>
</ul>
```

The resulting tree inserts a `<ul>` inside another `<ul>`, conflicting with [HTML](https://html.spec.whatwg.org/multipage/grouping-content.html#the-ul-element):

>Content model:
>    Zero or more li and script-supporting elements.

I guess we have two options:

* Change the indentation behavior: I don't think we want to change the behavior that has been there for decades, though. The previous author also [preferred the current behavior because of algorithm simplicity](https://w3c.github.io/editing/ActiveDocuments/execCommand.html#indenting-and-outdenting).
* Change HTML. This was [once rejected](https://www.w3.org/Bugs/Public/show_bug.cgi?id=12609) but IMO is the way to go for web compatibility.

Pinging @annevk for the HTML part.

**Comment by annevk** [comment]
Well, we could also do nothing or add `indentCorrectly` or some such.

Having said that, I recommend opening an issue against the HTML Standard to see what people think.

**Comment by rniwa** [comment]
> * Change the indentation behavior: I don't think we want to change the behavior that has been there for decades, though. The previous author also [preferred the current behavior because of algorithm simplicity](https://w3c.github.io/editing/ActiveDocuments/execCommand.html#indenting-and-outdenting).

We definitely can't change this behavior now. A lot of content depends on this behavior.

> * Change HTML. This was [once rejected](https://www.w3.org/Bugs/Public/show_bug.cgi?id=12609) but IMO is the way to go for web compatibility.

I support changing the HTML spec to allow this construct although it's not really a show stopper given all browsers interoperate anyway.

**Comment by masayuki-nakano** [comment]
I agree with @rniwa -san. `<li>` containing only sub list element may also have a bullet, and I don't think this style is better because word processors do not use this style.
